agent/consul/state/catalog.go

```diff
@@ -3652,6 +3653,18 @@ func updateGatewayNamespace(tx WriteTxn, idx uint64, service *structs.GatewaySer
 			continue
 		}
 
+		supportsIngress, supportsTerminating, err := serviceConnectInstances(tx, sn.ServiceName, entMeta)
+		if err != nil {
+			return err
+		}
+
+		if service.GatewayKind == structs.ServiceKindIngressGateway && !supportsIngress {
+			continue
+		}
+		if service.GatewayKind == structs.ServiceKindTerminatingGateway && !supportsTerminating {
```

Comment:
I'm not quiet sure I understand this. I'm wondering how that would work with destinations, which are service-defaults without service registration and that could be used with terminating-gateways?

Reply:
Do destinations get picked up by a wildcard entry in a gateway currently? If so, we'd need to make an exception somehow for services with a `service-defaults` config entry set.

Reply:
I added an exception for terminating gateways here when the service kind passed in is `GatewayServiceKindDestination` - that should mean the behavior of destinations won't be affected by this change.

Reply:
Shouldn't we add the check for `GatewayServiceKindDestination` in all the places we currently test on `structs.ServiceKindTerminatingGateway && !supportsTerminating`?